src/library/basesqltablemodel.cpp

```diff
@@ -714,6 +714,41 @@ QString BaseSqlTableModel::getTrackLocation(const QModelIndex& index) const {
     return QDir::fromNativeSeparators(nativeLocation);
 }
 
+CoverInfo BaseSqlTableModel::getCoverInfo(const QModelIndex& index) const {
+    CoverInfo coverInfo;
+    coverInfo.hash =
+            index.sibling(index.row(),
+                         fieldIndex(ColumnCache::
+                                         COLUMN_LIBRARYTABLE_COVERART_HASH))
+                    .data()
+                    .toUInt();
+    coverInfo.type = static_cast<CoverInfo::Type>(
+            index.sibling(index.row(),
+                         fieldIndex(ColumnCache::
+                                         COLUMN_LIBRARYTABLE_COVERART_TYPE))
+                    .data()
+                    .toInt());
+    coverInfo.source = static_cast<CoverInfo::Source>(
+            index.sibling(index.row(),
+                         fieldIndex(ColumnCache::
+                                         COLUMN_LIBRARYTABLE_COVERART_SOURCE))
+                    .data()
+                    .toInt());
+    coverInfo.coverLocation =
```

Comment:
This assignment is obviously wrong and causes a debug assertion:

```
    (assertion=assertion@entry=0xf90f90 "(coverInfo.type != CoverInfo::METADATA) || coverInfo.coverLocation.isEmpty()", file=file@entry=0xf9191a "../src/track/track.cpp", line=line@entry=1248, function=function@entry=0xf90f58 "void Track::setCoverInfo(const CoverInfoRelative&)") at /usr/include/qt5/QtCore/qlogging.h:90
#5  0x0000000000525721 in Track::setCoverInfo(CoverInfoRelative const&) (this=0x2e0c8270, coverInfo=...) at /usr/include/qt5/QtCore/qmutex.h:270
#6  0x00000000007fa0eb in BaseCoverArtDelegate::slotCoverFound(QObject const*, CoverInfo const&, QPixmap const&, unsigned long long, bool)
    (this=0xc4a5be0, pRequestor=<optimized out>, coverInfo=..., pixmap=..., requestedCacheKey=<optimized out>, coverInfoUpdated=<optimized out>)
```

Reply:
Please always test with debug assertions enabled and start Mixxx with --debugAssertBreak!

Reply:
Fixed! I only used --developer and didn't noticed any malfunction. --debugAssertBreak is the better option for debugging.